testsuite/MDAnalysisTests/coordinates/test_chainreader.py

```diff
@@ -455,3 +455,23 @@ def wrapped(ts):
     # in the issue report values of ~25 were seen before the fix
     # with the fix all values should be <1e-6
     assert np.abs(com).max() < 1e-6
+
+
+def test_issue_3349():
+    # test for repr of a chainreader when the subreaders are memoryreaders
+    cdx = np.random.random(3341 * 3 * 10)
+    cdx = cdx.reshape(-1, 3341, 3).astype(np.float32) * 10
+    u = mda.Universe(PSF, (cdx, DCD))
+    u2 = mda.Universe(PSF, (cdx, cdx, cdx, DCD))
+    print(u.trajectory.filenames)
```

Comment:
```suggestion
```
Thanks @ljwoods2, looks great! Just one final thing, perhaps let's not keep the print statement?